src/scripts/bsi_test_report.py

```diff
@@ -0,0 +1,227 @@
+
+    @property
+    def architecture(self):
+        return self.get_property('architecture')
+
+    @property
+    def build_target(self):
+        return self.get_property('ci_target')
+
+    @property
+    def timestamp(self):
+        # Timestamp is in fully qualified ISO format (potentially with) time zone offset.
+        # Python supports this only from 3.11
+        #
+        # See: https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat
+        if sys.version_info >= (3, 11):
+            return datetime.fromisoformat(self.get_property('timestamp'))
+        elif sys.version_info >= (3, 7):
+            return datetime.fromisoformat(self.get_property('timestamp')[:10])
+        else:
+            raise RuntimeError("Python is too old to read ISO timestamp")
+
+    @property
+    def assertions(self):
+        asserts = map(lambda x: int(
+            x._elem.attrib['assertions']), chain.from_iterable(self))
+        return sum(asserts)
+
+    def _format_command(self, command):
+        cmd = command.replace('\\', '/').split(' --')
+
+        def irrelevant_params(param):
+            filtered_params = ['prefix', 'report-properties',
+                               'with-external-includedir', 'test-results-dir']
+            for p in filtered_params:
+                if param.startswith(p):
+                    return False
+            return True
+        return '\n --'.join(filter(irrelevant_params, cmd))
+
+    def _escape_multiline_string(self, content, indentation_depth=0):
+        first_line = True
+        out = []
+        for line in content.split('\n'):
+            if first_line:
+                out.append('| %s' % line)
+                first_line = False
+            else:
+                out.append('%s| %s' % (' ' * indentation_depth, line))
+        return '\n'.join(out)
+
+    def render(self):
+        rst = '\n'.join([
+            '.. list-table:: %s %s %s' % (self.os, self.compiler, self.build_target),
+            '  :widths: 20 80',
+            '  :header-rows: 0'])
+        rst += '\n\n'
+
+        contents = {'Build Name': self.build_target,
+                    'Build Config': self._format_command(self.configure_command),
+                    'Test Execution': self._format_command(self.test_command),
+                    'Date': self.timestamp.date(),
+                    'System': self.os,
+                    'OS Version': self.os_version,
+                    'Compiler': self.compiler + ' ' + self.compiler_version,
+                    'Architecture': self.architecture,
+                    'Tests Executed': '%d (%d assertions)' % (self.tests, self.assertions),
+                    'Tests Passed': self.tests - self.failures - self.skipped,
+                    'Tests Failed': self.failures,
+                    'Tests Skipped': self.skipped}
+
+        for title, content in contents.items():
+            str_content = str(content)
+            if '\n' not in str_content:
```

Comment:
Maybe make this check inside `_escape_multiline_string()` itself, and here simply always call the escape function.

Reply:
Good point.